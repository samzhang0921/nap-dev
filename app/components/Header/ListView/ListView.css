.ListView {
  font-size: 11px;
  width: 102px;
  height: 20px;
  padding: 0;
  margin-top: 1px;
  margin-right: 8px;
  font-weight: 400;
  text-align: center;
  line-height: 20px;
}
