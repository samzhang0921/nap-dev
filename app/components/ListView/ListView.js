import React from 'react';
import {render} from 'react-dom';
import style from './ListView.css';

export default class ListView extends React.Component {
  constructor(props){
    super(props)
    this.state ={
      buttonClicked: false
    }

    this.changeListView = this.changeListView.bind(this);
  }

changeListView (buttonClicked){
  this.setState({
      buttonClicked: !buttonClicked
  })
  this.props.changeProductsView(!buttonClicked);
}

    render(){
      const buttonText = this.state.buttonClicked ? 'Outfit View' : 'Product View';
      return (
        <div >
          <button className={style.ListView} onClick = {()=> this.changeListView(this.state.buttonClicked)}>{buttonText}</button>
        </div>
      );
    }

}
