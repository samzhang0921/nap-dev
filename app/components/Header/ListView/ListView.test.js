import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import ListView from './ListView';

describe('ListView', () => {

  const changeProductsView = jest.fn();


  it('should match its snapshot with items', () => {

    const tree = renderer.create(<ListView changeProductsView = {changeProductsView}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change class on the onClick function', () => {

    const component = mount(<ListView changeProductsView = {changeProductsView}/>);
    component.find('.ListView').simulate('click');
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('button').text()).toBe('Outfit View');
  });
});
