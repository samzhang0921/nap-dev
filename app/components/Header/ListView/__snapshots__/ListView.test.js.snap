// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView should change class on the onClick function 1`] = `
<ListView
  changeProductsView={[Function]}
>
  <div>
    <button
      className="ListView"
      onClick={[Function]}
    >
      Outfit View
    </button>
  </div>
</ListView>
`;

exports[`ListView should match its snapshot with items 1`] = `
<div>
  <button
    className="ListView"
    onClick={[Function]}
  >
    Product View
  </button>
</div>
`;

exports[`ListView should mount item  1`] = `
<ListView>
  <div>
    <button
      className="ListView"
      onClick={[Function]}
    >
      Product View
    </button>
  </div>
</ListView>
`;
